server/find/http/server.go

```diff
@@ -170,6 +170,10 @@ func (s *Server) findCid(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) findMultihash(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodPost {
```

Comment:
This handler is associated with `/multihash/` pattern, i.e. `/mutlihash` subtree. 

This should immediately return `MethodNotAllowed` IMHO, because the presence of a tailing `/` means the client is in subtree teritory which does not support `POST` method and only supports `GET`.